packages/core/src/routes/sso-connector/index.ts

```diff
@@ -117,12 +117,17 @@ export default function singleSignOnRoutes<T extends AuthedRouter>(...args: Rout
   /* Get all single sign on connectors */
   router.get(
     pathname,
+    koaPagination({ isOptional: true }),
     koaGuard({
       response: ssoConnectorWithProviderConfigGuard.array(),
       status: [200],
     }),
     async (ctx, next) => {
-      const connectors = await getSsoConnectors();
+      const { limit, offset, disabled: paginationDisabled } = ctx.pagination;
+      const [totalCount, connectors] = paginationDisabled
+        ? await getSsoConnectors()
+        : await getSsoConnectors(limit, offset);
+      ctx.pagination.totalCount = totalCount;
```

Comment:
Feels like we could just always enable the pagination, keep the logic simple.

Reply:
Per offline discussion, in the current stage, our customers may not have too many SSO connectors but adding the pagination logic is simple here.